### PR TITLE
document rewind

### LINKED ISSUE #632: Using JxlDecoderSetPreferredColorProfile with JxlDecoderRewind
Hello,

I use `JxlDecoderSetPreferredColorProfile` in my programs.

I observed that after `JxlDecoderRewind`, the preferred color profile is forgotten, and the image data decoded later corresponds to the image's default color profile (like the JxlDecoderSetPreferredColorProfile call was never used).

I have to subscribe for the `JXL_DEC_COLOR_ENCODING` event again, because that's the only place where JxlDecoderSetPreferredColorProfile could be set again now.

Is it desired behavior?
If it is desired, please add info to documentation.
Or could be the JxlDecoderRewind call be made to remember the color profile preference?

### PR DESCRIPTION
**Pull request by mo271**
Fixes #632

### CODE REVIEW

File: lib/include/jxl/decode.h [hunk 1]
```diff
@@ -366,6 +366,15 @@ typedef enum {
...
  * and @ref JXL_DEC_COLOR_ENCODING, since they will provide the same information
  * as before.
+ * The difference to @ref JxlDecoderReset is that some state is kept, namely
+ * settings set by a call to
+ *  - @ref JxlDecoderSetCoalescing,
+ *  - @ref JxlDecoderSetDesiredIntensityTarget,
+ *  - @ref JxlDecoderSetDecompressBoxes,
+ *  - @ref JxlDecoderSetKeepOrientation,
+ *  - @ref JxlDecoderSetRenderSpotcolors,
+ *  - @ref JxlDecoderSkipFrames, and
```
Comment (szabadka) [inline lib/include/jxl/decode.h:376]: dec->skip_frames counter is reset when we rewind.

File: lib/include/jxl/decode.h [hunk 2]
```diff
@@ -366,6 +366,15 @@ typedef enum {
...
+ *  - @ref JxlDecoderSubscribeEvents.
```
Comment (szabadka) [inline lib/include/jxl/decode.h:377]: JxlDecoderSetParallelRunner should be added to the list.

**Review by szabadka — CHANGES_REQUESTED** [review]

File: lib/include/jxl/decode.h (hunk 1 above)
Comment (mo271) [reply lib/include/jxl/decode.h]: done

File: lib/include/jxl/decode.h (hunk 2 above)
Comment (mo271) [reply lib/include/jxl/decode.h:377]: done

**Review by szabadka — APPROVED** [review]

**Pull request merged**